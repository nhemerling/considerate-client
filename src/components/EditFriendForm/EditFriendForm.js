import React, { Component } from 'react';
import FriendApiService from '../../services/friend-api-service';
import moment from 'moment';
import './EditFriendForm.css';

export default class EditFriendForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      friend_name: '',
      occasion: '',
      occasion_date: '',
      error: null,
    };
  }

  componentDidMount() {
    const friendId = this.props.friendId;
    FriendApiService.getFriendById(friendId)
      .then((friend) => {
        this.setState({
          friend_name: friend.friend_name,
          occasion: friend.occasion,
          occasion_date: friend.occasion_date,
          error: null,
        });
      })
      .catch((res) => this.setState({ error: res.error }));
  }

  handleChange = (target) => {
    const { name, value } = target;
    this.setState({
      [name]: value,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const { friend_name, occasion, occasion_date } = this.state;
    const updateFriend = {
      friend_name: friend_name,
      occasion: occasion,
      occasion_date: occasion_date,
    };

    this.setState({ error: null });

    FriendApiService.updateFriend(this.props.friendId, updateFriend)
      .then(this.props.onEditFriendSuccess())
      .catch((res) => this.setState({ error: res.error }));
  };

  render() {
    const { error, friend_name, occasion, occasion_date } = this.state;
    return (
      <section className="EditFriendForm">
        <form className="EditFriendForm" onSubmit={this.handleSubmit}>
          <div role="alert">{error && <p className="red">{error}</p>}</div>
          <div className="friend_name">
            <label htmlFor="EditFriendForm__friend_name">Friend name</label>
            <input
              name="friend_name"
              type="text"
              required
              id="EditFriendForm__friend_name"
              value={friend_name}
              onChange={(e) => this.handleChange(e.target)}
            ></input>
          </div>
          <div className="occasion">
            <label htmlFor="EditFriendForm__occasion">Occasion</label>
            <input
              name="occasion"
              type="text"
              required
              id="EditFriendForm__occasion"
              value={occasion}
              onChange={(e) => this.handleChange(e.target)}
            ></input>
          </div>
          <div className="occasion_date">
            <label htmlFor="EditFriendForm__occasion_date">Occasion Date</label>
            <input
              name="occasion_date"
              type="date"
              required
              id="EditFriendForm__occasion_date"
              value={moment(occasion_date).format('YYYY-MM-DD')}
              onChange={(e) => this.handleChange(e.target)}
            ></input>
          </div>
          <button className="submit" type="submit">
            Submit
          </button>
        </form>
      </section>
    );
  }
}
