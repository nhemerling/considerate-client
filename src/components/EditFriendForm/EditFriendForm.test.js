import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditFriendForm from './EditFriendForm';

describe('EditFriendForm component', () => {
  const friendId = 0;

  it('renders an EditFriendForm by default', () => {
    const wrapper = shallow(<EditFriendForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders a specific EditFriendForm when given a friendId', () => {
    const wrapper = shallow(<EditFriendForm friendId={friendId} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
