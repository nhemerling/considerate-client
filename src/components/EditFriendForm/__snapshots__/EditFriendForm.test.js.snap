// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditFriendForm component renders a specific EditFriendForm when given a friendId 1`] = `
<section
  className="EditFriendForm"
>
  <form
    className="EditFriendForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="friend_name"
    >
      <label
        htmlFor="EditFriendForm__friend_name"
      >
        Friend name
      </label>
      <input
        id="EditFriendForm__friend_name"
        name="friend_name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="occasion"
    >
      <label
        htmlFor="EditFriendForm__occasion"
      >
        Occasion
      </label>
      <input
        id="EditFriendForm__occasion"
        name="occasion"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="occasion_date"
    >
      <label
        htmlFor="EditFriendForm__occasion_date"
      >
        Occasion Date
      </label>
      <input
        id="EditFriendForm__occasion_date"
        name="occasion_date"
        onChange={[Function]}
        required={true}
        type="date"
        value="Invalid date"
      />
    </div>
    <button
      className="submit"
      type="submit"
    >
      Submit
    </button>
  </form>
</section>
`;

exports[`EditFriendForm component renders an EditFriendForm by default 1`] = `
<section
  className="EditFriendForm"
>
  <form
    className="EditFriendForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="friend_name"
    >
      <label
        htmlFor="EditFriendForm__friend_name"
      >
        Friend name
      </label>
      <input
        id="EditFriendForm__friend_name"
        name="friend_name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="occasion"
    >
      <label
        htmlFor="EditFriendForm__occasion"
      >
        Occasion
      </label>
      <input
        id="EditFriendForm__occasion"
        name="occasion"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="occasion_date"
    >
      <label
        htmlFor="EditFriendForm__occasion_date"
      >
        Occasion Date
      </label>
      <input
        id="EditFriendForm__occasion_date"
        name="occasion_date"
        onChange={[Function]}
        required={true}
        type="date"
        value="Invalid date"
      />
    </div>
    <button
      className="submit"
      type="submit"
    >
      Submit
    </button>
  </form>
</section>
`;
